draco/analysis/transform.py

```diff
@@ -1265,3 +1257,213 @@ def process_finish(self) -> Union[containers.SiderealStream, containers.RingMap]
+        -------
+        out
+            Dataset of same type as input with axes reduced. Any datasets
+            which are not included in the reduction list will not be initialized,
+            other than weights.
+        """
+        out = self._make_output_container(data)
+        out.add_dataset(self.dataset)
+
+        # Get the dataset
+        ds = data.datasets[self.dataset]
+        original_ax_id = ds.distributed_axis
+
+        # Get the axis indices to apply the operation over
+        ds_axes = list(ds.attrs["axis"])
+
+        # Get the new axis to distribute over
+        if ds_axes[original_ax_id] not in self.axes:
+            new_ax_id = original_ax_id
+        else:
+            ax_priority = [
+                x for _, x in sorted(zip(ds.shape, ds_axes)) if x not in self.axes
+            ]
+            if not ax_priority:
+                raise ValueError(
+                    "Could not find a valid axis to redistribute. At least one "
+                    "axis must be omitted from filtering."
+                )
+            # Get the longest axis
+            new_ax_id = ds_axes.index(ax_priority[-1])
+
+        new_ax_name = ds_axes[new_ax_id]
+
+        # Redistribute the dataset to the target axis
+        ds.redistribute(new_ax_id)
+        # Redistribute the output container (group) to the target axis
+        # Since this is a container, distribute based on axis name
+        # rather than index
+        out.redistribute(new_ax_name)
+
+        # Get the weights
+        if hasattr(data, "weight"):
+            weight = data.weight[:]
+            # The weights should be distributed over the same axis as the array,
+            # even if they don't share all the same axes
+            new_weight_ax = list(data.weight.attrs["axis"]).index(new_ax_name)
+            weight = weight.redistribute(new_weight_ax)
+        else:
+            self.log.info("No weights available. Using equal weighting.")
+            weight = np.ones(ds.local_shape, ds.dtype)
+
+        # Apply the reduction, ensuring that the weights have the correct dimension
+        weight = np.broadcast_to(weight, ds.local_shape, subok=False)
+        apply_over = tuple([ds_axes.index(ax) for ax in self.axes if ax in ds_axes])
+
+        reduced, reduced_weight = self.reduction(
+            ds[:].local_array[:], weight, apply_over
+        )
+
+        # Add the reduced data and redistribute the container back to the
+        # original axis
+        out[self.dataset][:] = reduced[:]
+
+        if hasattr(out, "weight"):
+            out.weight[:] = reduced_weight[:]
+
+        # Redistribute bcak to the original axis, again using the axis name
+        out.redistribute(ds_axes[original_ax_id])
+
+        return out
+
+    def _make_output_container(
+        self, data: containers.ContainerBase
+    ) -> containers.ContainerBase:
+        """Create the output container."""
+        output_axes = {ax: np.array([-1], dtype=int) for ax in self.axes}
```

Comment:
Why -1? Something like the mean of the reduced axis might be more meaningful.

Reply:
This was just an arbitrary choice to make a length 1 axis that explicitly doesn't have meaning. I'm not convinced that something like a mean would be better - for example, if we took a variance over RA then the RA axis mean would be 180 deg, and I'm not sure if that's really meaningful (and maybe confusing since the data wouldn't actually correspond to that position). It would probably be better to use something a bit more explicit though.

Reply:
Swapped to using the first value of the original index map by default after some discussion